### ISSUE TITLE
Option to warn when logging format string is not a string literal

### ISSUE THREAD

**Issue by tapted** [pylint-dev/pylint#8848]
### Current problem

One may write working code, something akin to

```py
errors = f"{sys.argv[1]} some-error %s"
logging.error(errors, context)
```

but if argv[1] ever contains a `%s` of its own (e.g. `file%s.bar`), a runtime exception is raised:

```
TypeError: not enough arguments for format string
Call stack:
    logging.error(errors, context)
Message: 'one%s some-error %s'
```

The code could be written as

```py
logging.error("%s some-error %s, sys.argv[1], context)
```

or

```py
errors = f"{sys.argv[1]} some-error {context}"
logging.error(errors)
```

or (better?)

```py
errors = f"{sys.argv[1]} some-error {context}"
logging.error("%s", errors)
```

There may be a debate about whether the second example should _also_ trip a lint. `errors` contains format specifiers, but if `logging.error` is invoked with only a single argument, it seems "safe": no TypeError is raised in this case. However, gcc's `-Wformat-nonliteral` (which this would be similar to) would certainly complain.

So the case where logging methods are called with only a single argument may be exempt from this lint, or opt-in -- seems more of a stylistic preference.

But if that's the case, the example is admittedly a little contrived.. It's odd to separate the format specifiers into the `errors` variable. And once the substitutions move to the argument to `logging.error`, they will trip the `logging-not-lazy` lint which already exists.

So maybe nothing is needed. Or maybe it's worth encouraging `logging.error("%s", errors)` over `logging.error(errors)` for the single-argument case for consistency.

### Desired solution

A new lint. E.g., `logging-format-not-literal`
